codegen/src/main/java/software/amazon/awssdk/codegen/poet/auth/scheme/DefaultAuthSchemeParamsSpec.java

```diff
@@ -119,23 +136,52 @@ private void addFieldsAndAccessors(TypeSpec.Builder b) {
                                   .addStatement("return region == null ? Optional.empty() : Optional.of(region)")
                                   .build());
         }
+
+        if (authSchemeSpecUtils.generateEndpointBasedParams()) {
+            parameters().forEach((name, model) -> {
+                if (authSchemeSpecUtils.includeParam(model, name)) {
+                    String methodName = authSchemeSpecUtils.paramMethodName(name);
+                    TypeName typeName = parameterType(model);
+                    b.addField(FieldSpec.builder(typeName, methodName)
+                                        .addModifiers(Modifier.PRIVATE, Modifier.FINAL)
+                                        .build());
+                    MethodSpec.Builder spec = authSchemeSpecUtils.endpointParamAccessorSignature(model, name)
+                                                                 .addAnnotation(Override.class);
+                    b.addMethod(spec.addStatement("return " + methodName)
+                                    .build());
```

Comment:
nit: could be one line. and even combined with above line `b.addMethod(autheSchemeSpecUtils....`